ui/appui-react/src/appui-react/accudraw/AccuDrawFieldContainer.tsx

```diff
@@ -206,7 +213,30 @@ export function AccuDrawFieldContainer(props: AccuDrawFieldContainerProps) {
     []
   );
 
+  const focusNextField = React.useCallback((field: ItemField) => {
+    switch (field) {
+      case ItemField.X_Item:
+        IModelApp.accuDraw.setFocusItem(ItemField.Y_Item);
+        break;
+      case ItemField.Y_Item:
+        showZ ?
+          IModelApp.accuDraw.setFocusItem(ItemField.Z_Item) :
+          IModelApp.accuDraw.setFocusItem(ItemField.X_Item);
+        break;
+      case ItemField.Z_Item:
+        IModelApp.accuDraw.setFocusItem(ItemField.X_Item);
+        break;
+      case ItemField.ANGLE_Item:
+        IModelApp.accuDraw.setFocusItem(ItemField.DIST_Item);
+        break;
+      case ItemField.DIST_Item:
+        IModelApp.accuDraw.setFocusItem(ItemField.ANGLE_Item);
+        break;
+    }
+  }, [showZ]);
+
   const handleEscPressed = React.useCallback(() => {
+    IModelApp.accuDraw.unlockAllFields();
```

Comment:
I understand the focus out on ESC. Wondering the reason for unlocking all the fields too?

Reply:
Usually, `Escape` is used to cancel the tool. So we thought that using `Escape` to cancel AccuDraw inputs might be the natural reflex.

Reply:
If ESC cancels the tool, then unlocking the fields isn't necessary. If ESC is not canceling the tool, then I think it should leave the fields locked. If the user has a field locked but pressed ESC to set focus to tool settings or something, I think unlock the field would be unexpected...

Reply:
Alright, I'll remove it, anyway we can add it back at some point, if needed.